src/epstats/toolkit/experiment.py

```diff
@@ -169,6 +175,15 @@ def __init__(
         self.filters = filters if filters is not None else []
         self.outlier_detection_algorithm = outlier_detection_algorithm
 
+    def _check_metric_ids_unique(self):
+        """
+        Raises an exception if `metrics` contain duplicated ids.
+        """
+        id_counts = Counter(metric.id for metric in self.metrics)
+        for id_, count in id_counts.items():
+            if count > 1:
+                raise ValueError(f"Metric ids must be unique. Id={id_} found more than once.")
```

Comment:
Those are ids from EP UI? Because I believe we do have duplicate ids if it's metric with attribute (like CTR)

Reply:
No these are just internal ep-stats ids. In the request they are always sequential starting from 1.